server/common/fbs/fbs_coders.py

```diff
@@ -0,0 +1,225 @@
+import numpy as np
+import pandas as pd
+from scipy import sparse as sp
+import json
+
+from server.common.utils.type_conversion_utils import get_encoding_dtype_of_array
+import server.common.fbs.NetEncoding.TypedFBArray as TypedFBArray
+import server.common.fbs.NetEncoding.Float32FBArray as Float32FBArray
+import server.common.fbs.NetEncoding.Float64FBArray as Float64FBArray
+import server.common.fbs.NetEncoding.Int32FBArray as Int32FBArray
+import server.common.fbs.NetEncoding.JSONEncodedFBArray as JSONEncodedFBArray
+import server.common.fbs.NetEncoding.Uint32FBArray as Uint32FBArray
+import server.common.fbs.NetEncoding.DictEncoded16FBArray as DictEncoded16FBArray
+import server.common.fbs.NetEncoding.DictEncoded32FBArray as DictEncoded32FBArray
+import server.common.fbs.NetEncoding.DictEncoded8FBArray as DictEncoded8FBArray
+import server.common.fbs.NetEncoding.SparseFloat32FBArray as SparseFloat32FBArray
+import server.common.fbs.NetEncoding.SparseFloat64FBArray as SparseFloat64FBArray
+
+
+def _get_array_class(array):
+    # return the generic type of array - category, sparse, or dense
+    # this determines which coder should be used.
+    if isinstance(array, pd.Series) and array.dtype.name == "category":
```

Comment:
Are you specifically excluding the more exotic types such as Panda CategoricalIndex?

You might consider using the `pandas.api.types` helpers that make the intent more obvious, eg, [pandas.api.types.is_categorical_dtype](https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_categorical_dtype.html), or pandas.api.types.is_categorical (depending on your intent).